lib/active_merchant/billing/credit_card_methods.rb

```diff
@@ -241,6 +242,11 @@ module CreditCardMethods
         3528..3589, 3088..3094, 3096..3102, 3112..3120, 3158..3159, 3337..3349
       ]
 
+      HIPERCARD_RANGES = [
+        384100..384100, 384140..384140, 384160..384160, 606282..606282, 637095..637095,
+        637568..637568, 637599..637599, 637609..637609, 637612..637612
+      ]
+
```

Comment:
Do we want to add `HIPERCARD` ranges since they are a subset of mastercard?

Reply:
The subset of mastercard are some of the ones that start with 6. The ones that don't fall within the `MASTERCARD_RANGES`.

Reply:
Hmmmm I'm just thinking, does AM perform any check to see if the bin for the card is in a gateway's support card list?

Reply:
Core does based on AM, if the bin isn't found then no type will be assigned which makes the PM invalid

Reply:
So do we think there's a scenario where core sends in a bin thinking it's mastercard, gets back hipercard, then it attempts to send to the gateway and it fails because the gateway doesn't support it?

Reply:
That wouldn't happen because of the order of events, MasterCard is on the top `CARD_COMPANY_DETECTORS`.

Reply:
606282, 637095, 637568, 637599, 637609 are the ones that match the Mastercard bin ranges. And those bin ranges were only added 22 months to Mastercard, https://github.com/activemerchant/active_merchant/commit/db68b54d2942d9b9a26a3d9a65b2728794071a18